### PR TITLE
[16.0][MIG] base_ebill_payment_contract

### PR DESCRIPTION
**Pull request by TDu**

### CODE REVIEW

**Review by sebalix — APPROVED** [review]

File: base_ebill_payment_contract/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        self.ensure_one()
+        base_domain = [
+            ("is_valid", "=", True),
+            ("partner_id", "=", self.id),
+            ("transmit_method_id", "=", transmit_method.id),
+        ]
+        contracts = self.env["ebill.payment.contract"].search(
+            AND([domain or [], base_domain]), limit=1
+        )
+        if not contracts:
```
Comment (grindtildeath) [inline base_ebill_payment_contract/models/res_partner.py]:
```suggestion
        contract = self.env["ebill.payment.contract"].search(
            AND([domain or [], base_domain]), limit=1
        )
        if not contract:
```
Nitpicking: you're going to have max 1 contract with `limit=1`

**Review by grindtildeath — APPROVED** [review]

File: base_ebill_payment_contract/models/res_partner.py (hunk 1 above)
Comment (TDu) [reply base_ebill_payment_contract/models/res_partner.py]: Updated

File: base_ebill_payment_contract/tests/test_base_ebill_payment_contract.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from datetime import date, timedelta
+
+from odoo.tests.common import Form, SingleTransactionCase
+
+
+class TestCrmOpportunityCurrency(SingleTransactionCase):
```
Comment (simahawk) [inline base_ebill_payment_contract/tests/test_base_ebill_payment_contract.py:9]: CRM opportunity? :smile: 
Leave it like this... we can change it after the merge.

Comment (TDu) [reply base_ebill_payment_contract/tests/test_base_ebill_payment_contract.py:9]: Fixed with https://github.com/OCA/edi/pull/981

**Pull request merged**